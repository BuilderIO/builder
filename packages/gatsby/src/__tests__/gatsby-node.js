jest.mock(`graphql-tools`, () => {
  return {
    transformSchema: jest.fn(),
    introspectSchema: jest.fn(),
    RenameTypes: jest.fn(),
  }
})
jest.mock(`apollo-link-http`, () => {
  return {
    createHttpLink: jest.fn(),
  }
})
const { createHttpLink } = require(`apollo-link-http`)
jest.mock(`gatsby/graphql`, () => {
  const graphql = jest.requireActual(`gatsby/graphql`)
  return {
    ...graphql,
    buildSchema: jest.fn(),
    printSchema: jest.fn(),
  }
})
const { sourceNodes, createPages } = require(`../gatsby-node`)
const nodeFetch = require(`node-fetch`)
const mockResponse = {
  data: {
    allBuilderModels: {
      page: [],
    },
  },
}
const getInternalGatsbyAPI = () => {
  const actions = {
    addThirdPartySchema: jest.fn(),
    createNode: jest.fn(),
    createPage: jest.fn(),
  }

  return {
    actions,
    cache: {
      get: jest.fn(),
      set: jest.fn(),
    },
    createContentDigest: jest.fn(),
    createNodeId: jest.fn(),
    graphql: jest.fn(() => Promise.resolve(mockResponse)),
  }
}

describe(`validation`, () => {
  it(`throws on missing public API Key`, () => {
    expect(
      sourceNodes(getInternalGatsbyAPI(), { templates: {} })
    ).rejects.toThrowErrorMatchingSnapshot()
  })

  it(`throws on an invalid templates path`, () => {
    expect(
      createPages(getInternalGatsbyAPI(), {
        templates: { page: `Invalid` },
        publicAPIKey: `mock___`,
      })
    ).rejects.toThrowErrorMatchingSnapshot()
  })
})

describe(`createSchemaNode`, () => {
  it(`invokes createContentDigest`, async () => {
    const api = getInternalGatsbyAPI()
    await sourceNodes(api, {
      publicAPIKey: `mock___`,
    })
    expect(api.createContentDigest).toHaveBeenCalledTimes(1)
  })
})

describe(`createHttpLink`, () => {
  it(`use node fetch`, async () => {
    const api = getInternalGatsbyAPI()
    await sourceNodes(api, {
      publicAPIKey: `mock__`,
    })

    expect(createHttpLink).toHaveBeenCalledWith(
      expect.objectContaining({ fetch: nodeFetch })
    )
  })
})
