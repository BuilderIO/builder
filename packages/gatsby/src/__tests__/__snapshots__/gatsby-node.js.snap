// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation throws on an invalid templates path 1`] = `"@builder.io/gatsby requires a valid template path for each model"`;

exports[`validation throws on missing public API Key 1`] = `"@builder.io/gatsby requires a public API Key"`;
